library/nmstate_l3_interface.py

```diff
@@ -120,14 +116,29 @@ def set_ipv4_addresses(interface_state, ipv4, purge=False):
     else:
         addresses = ipconfig.setdefault('addresses', [])
 
-    ip, prefix = ipv4.split('/')
-    addr = {'ip': ip, 'prefix-length': int(prefix)}
+    addr = create_ip_dict(ipv4)
     if addr not in addresses:
```

Comment:
EAFP applies here.
https://docs.python.org/3/glossary.html#term-eafp

Reply:
There is no exception that is avoided here, it is just a check to see if the address needs to be added, EAFP would be the  next comment with "try don't ask".

Reply:
Ahhh right, don't know why I wrote the comment here. Sorry.